src/ralph/virtual/tests/test_models.py

```diff
@@ -0,0 +1,88 @@
+from ralph.assets.models.assets import ServiceEnvironment
+from ralph.assets.models.choices import ComponentType
+from ralph.assets.models.components import ComponentModel, GenericComponent
+from ralph.assets.tests.factories import EnvironmentFactory, ServiceFactory
+from ralph.tests import RalphTestCase
+from ralph.virtual.models import (
+    CloudFlavor,
+    CloudHost,
+    CloudProject,
+    CloudProvider
+)
+
+
+class OpenstackModelsTestCase(RalphTestCase):
+    def setUp(self):
+        self.envs = EnvironmentFactory.create_batch(1)
+        self.services = ServiceFactory.create_batch(1)
+        self.service_env = ServiceEnvironment.objects.create(
+            service=self.services[0], environment=self.envs[0]
+        )
+
+        self.cloud_provider = CloudProvider.objects.create(name='openstack')
+        flavor1 = CloudFlavor.objects.create(
+            name='flavor_1',
+            flavor_id='flavor_id_1',
+            cloudprovider=self.cloud_provider
+        )
+
+        self.test_cpu = ComponentModel.objects.create(
+            name='vcpu1',
+            cores=4,
+            family='vCPU',
+            type=ComponentType.processor,
+        )
+
+        self.test_mem = ComponentModel.objects.create(
+            name='1024 MiB vMEM',
+            size='1024',
+            type=ComponentType.memory,
+        )
+        self.test_disk = ComponentModel.objects.create(
+            name='4 GiB vDISK',
+            size='4096',
+            type=ComponentType.disk,
+        )
+
+        GenericComponent.objects.create(asset=flavor1, model=self.test_cpu)
+        GenericComponent.objects.create(asset=flavor1, model=self.test_mem)
+        GenericComponent.objects.create(asset=flavor1, model=self.test_disk)
+
+        project1 = CloudProject.objects.create(
+            name='project1',
+            cloudprovider=self.cloud_provider,
+            project_id='project_id_1',
+        )
+
+        CloudHost.objects.create(
+            host_id='host_id1',
+            hostname='host1',
+            parent=project1,
+            cloudflavor=flavor1
+        )
+
+    def test_check_cloudflavor_cores(self):
```

Comment:
use ``ddt`` for ``test_check_cloudflavor_*`` ([example of use](https://github.com/allegro/ralph/blob/ng/src/ralph/data_center/tests/test_models.py#L199-L214))